presto-main/src/main/java/io/prestosql/event/QueryMonitor.java

```diff
@@ -209,6 +218,9 @@ private QueryStatistics createQueryStatistics(QueryInfo queryInfo)
             operatorSummaries.add(operatorStatsCodec.toJson(summary));
         }
 
+        Optional<StatsAndCosts> planNodeStatsAndCosts = queryInfo.getOutputStage().map(QueryMonitor::reconstructStatsAndCosts);
+        Optional<String> serializedPlanNodeStatsAndCosts = planNodeStatsAndCosts.map(statsAndCostsCodec::toJson);
```

Comment:
StatsAndCost is not a public API, so the serialized form can change at any time without warning.

Reply:
That's true, but QueryStatistcs also contains operatorSummaries, which is serialized OperatorStats internal structure. It is still helpful though (and required piece to do estimates vs stats comparison).

I plan those object to be dumped to storage for later analysis and such tools should be clever enough to deal with potentially various formats (e.g to do historical comparison or audit).

Reply:
maybe we always add presto version to QueryInfo / query json?